analysis/tests/src/CompletionFromModule.res

```diff
@@ -0,0 +1,39 @@
+module SomeModule = {
+  type t = {name: string}
+
+  @get external getName: t => string = "name"
+
+  let thisShouldNotBeCompletedFor = () => "hi"
+}
+
+let n = {SomeModule.name: "hello"}
+
+// ^dv+
+// n.
+//   ^com
+// ^dv-
+
+@mainTypeForModule(SomeOtherModule)
+type typeOutsideModule = {nname: string}
+
+module SomeOtherModule = {
+  type t = typeOutsideModule
```

Comment:
Is this alias as `t` a hard requirement to make this work?
Being newish to ReScript `type t` is a little strange and it feels a bit like an artificial limitation right now.